### PR TITLE
[onewiregpio] Added precision parameter to OneWireGpio Binding

### PR DESCRIPTION
**Pull request by theater**
Upon a request from a friend of mine I've added additional parameter to the binding which allows you to lower the precision of the received temperature number from the filesystem. 

A bit of background about the binding itself - the binding reads files inside /sys/bus/w1/devices/DEVICE_ID_FOLDER/w1_slave. 
If there is t=xxxxx, where xxxxx is integer value. Then the binding makes a BigDecimal.valueOf(xxxxx).movePointLeft(3) and returns the value. This makes tha maximum possible precision after the floating point to be total of 3 (i.e. the digits after the floating point). 
The precision parameter allows you to reduce this number down to 0, 1, 2 or 3(the default value is the max value) with round half up rounding strategy. This allows you not to spam event bus on every small change of the sensors and disables triggering of different rules without need (with too small changes of the number).

Saw that the original author is Anathol Ogorek but was not able to request a review from him as I couldn't find him on the Reviewers list. :( 

The changes from the commit:
* Added precision parameter which allows to lower precision of value
read from the file system
* Implemented configuration class for easier read of configuration
through the reflection method in the framework
* Moved the read of configuration parameters to initialize method
leaving the checkConfiguration method to do only validation
* Updated the README

Cheers,
Konstantin

### CODE REVIEW

File: bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -22,6 +22,13 @@
 				<default>600</default>
 			</parameter>
 
+			<parameter name="precision" type="integer">
```
Comment (martinvw) [inline bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml]: Please add the bounds also here

File: bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -22,6 +22,13 @@
...
+				<label>Sensor precision</label>
```
Comment (martinvw) [inline bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml]: Please capitalize the P

File: bundles/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/internal/handler/OneWireGPIOHandler.java [hunk 3]
```diff
@@ -84,21 +93,25 @@ public void initialize() {
...
-        if (refreshTime.intValue() <= 0) {
-            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}.", refreshTime,
+        if (refreshTime <= 0) {
+            logger.debug("Refresh time [{}] is not valid. Falling back to default value: {}.", refreshTime,
                     DEFAULT_REFRESH_TIME);
             refreshTime = DEFAULT_REFRESH_TIME;
         }
+
+        if (precision < 0 || precision > MAX_PRECISION_VALUE) {
+            logger.debug("Precision value {} is outside allowed values [0 - 3]. Falling back to default value: {}.",
```
Comment (martinvw) [inline bundles/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/internal/handler/OneWireGPIOHandler.java]: Consider using the constant 3 as well here to keep the logging correct when the constant is changes in the future

File: bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml (hunk 1 above)
Comment (theater) [reply bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml]: If I add bounds here, do I have to validate that the number is in these bounds in the code?

Comment (martinvw) [reply bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml]: I think it makes sense to keep the bounds there, I'm in doubt whether it was used only for the UI or also enforced for textual configuration.

Comment (theater) [reply bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml]: Hm... I just looked into the xsd and I'm not quite sure how to do that. Is it through options and writing down the possible values as string with limitToOptions=true afterwards or there is easier way to say it's integer with bounds 0-3?

**Comment by theater** [comment]
Did the boundaries with options. It appears that the option values are actually BigDecimal and not String. Probably string is only UI presentation. Cheers !

File: bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml (hunk 1 above)
Comment (martinvw) [reply bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml]: Sorry I was a little bit sick the last few days so I failed to answer :-)

https://github.com/openhab/openhab-addons/blob/88c489091862784167ce0880c4fbde2d8a0b6c0c/bundles/org.openhab.binding.sensebox/src/main/resources/ESH-INF/config/config-description.xml#L15

Just provide `min` and `max`

File: bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 4]
```diff
@@ -22,6 +22,20 @@
 				<default>600</default>
 			</parameter>
 
+			<parameter name="precision" type="integer">
+				<label>Sensor Precision</label>
+				<description>Sensor precision after floating point.</description>
+				<required>false</required>
+				<options>
+					<option value="0">0</option>
```
Comment (martinvw) [inline bundles/org.openhab.binding.onewiregpio/src/main/resources/ESH-INF/thing/thing-types.xml]: Options are only meant for string parameters

**Comment by martinvw** [comment]
Thanks for your quick response!

**Comment by theater** [comment]
No problem! Thanks for the review!  Hope you're well now. :)

**Pull request merged**